### PR TITLE
Flat

### PR DESCRIPTION
**Pull request by Smantii**

### CODE REVIEW

File: src/dctkit/dec/flat.py [hunk 1]
```diff
@@ -2,10 +2,12 @@
 from dctkit.dec import cochain as C
 from jax import Array, vmap
 from functools import partial
+from typing import Callable, Dict, Optional
 
 
-def flat(c: C.CochainP0 | C.CochainD0, weights: Array,
-         edges: C.CochainP1V | C.CochainD1V) -> C.CochainP1 | C.CochainD1:
+def flat(c: C.CochainP0 | C.CochainD0, weights: Array, edges: C.CochainP1V |
+         C.CochainD1V, weighted_I: Optional[Callable] = None,
```
Comment (alucantonio) [inline src/dctkit/dec/flat.py:9]: weighted_I sounds like "weighted_Identity": the name is misleading. Why not calling it "interp_fun"? What is the meaning of "weighted" here?

File: src/dctkit/dec/cochain.py [hunk 2]
```diff
@@ -394,3 +394,76 @@ def sym(c: Cochain) -> Cochain:
...
+    K = K.at[:n - kernel_window + 1, :].set(K_non_zero)
+    kernel_coch = Cochain(c.dim, c.is_primal, c.complex, K)
+
+    # apply hodge star
+    star_kernel = star(kernel_coch)
+    conv = Cochain(c.dim, c.is_primal, c.complex, star_kernel.coeffs@c.coeffs)
+    return conv
+
+
+def constant_sub(k: float, c: Cochain) -> Cochain:
```
Comment (alucantonio) [inline src/dctkit/dec/cochain.py:433]: I don't like this. A constant cochain is not special, we don't need a specific operation for this. If you need a wrapper, use it in the code of the specific application you are developing, not as a part of the dctkit library.

File: src/dctkit/dec/cochain.py [hunk 3]
```diff
@@ -394,3 +394,76 @@ def sym(c: Cochain) -> Cochain:
...
+
+    # generate a wider array that we want a slice into
+    buffer = buffer.at[:, :n].set(kernel.coeffs[:n].T)
+    buffer = buffer.at[:, n:].set(kernel.coeffs[:n-1].T)
+
+    rolled = buffer.reshape(-1)[n-1:-1].reshape(n, -1)
+    K_full_roll = jnp.roll(rolled[:, :n], shift=1, axis=0)
+    K_non_zero = K_full_roll[:n - kernel_window + 1]
+    K = K.at[:n - kernel_window + 1, :].set(K_non_zero)
+    kernel_coch = Cochain(c.dim, c.is_primal, c.complex, K)
```
Comment (alucantonio) [inline src/dctkit/dec/cochain.py:425]: We pass the argument "kernel" to the function. "Kernel" is a cochain. Here we define kernel_coch: what should we expect from this variable? Wasn't the kernel a cochain already? This part of the code is not clear: there are no comments and the names of the variables do not help understanding what it does.

**Review by alucantonio — CHANGES_REQUESTED** [review]

File: src/dctkit/dec/cochain.py (hunk 2 above)
Comment (Smantii) [reply src/dctkit/dec/cochain.py]: Removed.

File: src/dctkit/dec/flat.py (hunk 1 above)
Comment (Smantii) [reply src/dctkit/dec/flat.py]: Done

**Review by alucantonio — APPROVED** [review]

**Pull request merged**